src/server/implementation/users.ts

```diff
@@ -54,13 +54,23 @@ async function defaultCreateOrUpdateUser(
     existingSessionId,
     args,
   });
-  const existingUserId = existingAccount?.userId ?? null;
+  const existingUserId = (existingAccount?.userId ??
+    null) as GenericId<"users"> | null;
   if (config.callbacks?.createOrUpdateUser !== undefined) {
     logWithLevel(LOG_LEVELS.DEBUG, "Using custom createOrUpdateUser callback");
     return await config.callbacks.createOrUpdateUser(ctx, {
       existingUserId,
       ...args,
     });
+  } else {
+    if (
+      existingUserId &&
+      ctx.db.normalizeId("users", existingUserId) === null
+    ) {
+      throw new Error(
+        `User ID \`${existingUserId}\` is not in the \`users\` table`,
```

Comment:
Could use a better message about defining your own `createOrUpdateUser` to use another table